infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/UniqueNamesProvisioner.java

```diff
@@ -95,7 +97,7 @@ public void provision(T k8sEnv, RuntimeIdentity identity) throws InfrastructureE
     k8sEnv.getIngresses().clear();
     for (Ingress ingress : ingresses) {
       final ObjectMeta ingressMeta = ingress.getMetadata();
-      putLabel(ingress, Constants.CHE_ORIGINAL_NAME_LABEL, ingressMeta.getName());
+      putAnnotation(ingress, Constants.CHE_ORIGINAL_NAME_LABEL, ingressMeta.getName());
```

Comment:
It may be confusing that annotation is stored with key `CHE_ORIGINAL_NAME_LABEL`. Maybe it would be better to duplicate `CHE_ORIGINAL_NAME_LABEL` and have a separated constant for `CHE_ORIGINAL_NAME_ANNOTATION`.

Reply:
Not applicable anymore. We don't use annotations now..